chef/cookbooks/provisioner/libraries/repositories.rb

```diff
@@ -17,111 +17,27 @@
 class Provisioner
   class Repositories
     class << self
-      def suse_optional_repos(version, type)
-        case type
-        when :common
-          %w(PTF)
-        when :cloud
-          case version
-          when "11.3", "11.4"
-            %w(
-              Cloud
-              SUSE-OpenStack-Cloud-SLE11-6-Pool
-              SUSE-OpenStack-Cloud-SLE11-6-Updates
-            )
-          when "12.0"
-            []
-          when "12.1"
-            %w(
-              Cloud
-              SUSE-OpenStack-Cloud-6-Pool
-              SUSE-OpenStack-Cloud-6-Updates
-            )
-          else
-            []
-          end
-        when :hae
-          case version
-          when "11.3"
-            %w(SLE11-HAE-SP3-Pool SLE11-HAE-SP3-Updates)
-          when "11.4"
-            %w(SLE11-HAE-SP4-Pool SLE11-HAE-SP4-Updates)
-          when "12.0"
-            %w(SLE12-HA-Pool SLE12-HA-Updates)
-          when "12.1"
-            %w(SLE12-SP1-HA-Pool SLE12-SP1-HA-Updates)
-          else
-            []
-          end
-        when :storage
-          case version
-          when "12.0"
-            %w(SUSE-Enterprise-Storage-2-Pool SUSE-Enterprise-Storage-2-Updates)
-          else
-            []
-          end
-        end
-      end
-
-      def suse_get_repos_from_attributes(node,platform,version)
-        repos = Mash.new
-
-        if node[:provisioner][:suse] && node[:provisioner][:suse][:autoyast] && node[:provisioner][:suse][:autoyast][:repos]
-          if node[:provisioner][:suse][:autoyast][:repos][:common]
-            repos = node[:provisioner][:suse][:autoyast][:repos][:common].to_hash
-          end
-          product = "#{platform}-#{version}"
-          if node[:provisioner][:suse][:autoyast][:repos][product]
-            repos.merge! node[:provisioner][:suse][:autoyast][:repos][product].to_hash
-          end
-        end
-
-        repos
-      end
-
       def inspect_repos(node)
         unless node.roles.include? "provisioner-server"
           raise "Internal error: inspect_repos method should only be called on provisioner-server node."
         end
 
+        common_available, cloud_available, hae_available, storage_available = nil
```

Comment:
Actually, this all thing should be dropped: the attributes are needed because the rails app had no knowledge about repos before; now that it's fixed, we can just as well replace the use of these attributes by direct access to the repos data.